src/main/java/org/csanchez/jenkins/plugins/kubernetes/PodTemplateBuilder.java

```diff
@@ -184,7 +184,7 @@ public Pod build() {
         // Build volumes and volume mounts.
         Map<String, Volume> volumes = new HashMap<>();
         Map<String, VolumeMount> volumeMounts = new HashMap<>();
-
+        String podName = agent != null ? agent.getPodName() : null;
```

Comment:
Are you sure it's not an issue that podName can be null ? I wonder if it's not better to put a default name like "undetermined" or fail fast and throw an exception.